Userland/Services/WindowServer/Window.h

```diff
@@ -336,6 +336,12 @@ class Window final : public Core::Object
     const MenuBar* menubar() const { return m_menubar; }
     void set_menubar(MenuBar*);
 
+    template<typename EventHandler>
+    void set_event_handler(EventHandler&& event_handler)
+    {
+        m_event_handler = move(event_handler);
+    }
```

Comment:
I added a general mechanism to `Core::Object` for this so we don't have to specialize the behavior in Window :)

`Core::Object::set_event_filter(Function<bool(Event&)>)`

You set a filter on the target object, and then the filter gets consulted during event dispatch. Return `false` from the filter and the event is not delivered.

Reply:
Ah, didn't know, will look into it, thx.